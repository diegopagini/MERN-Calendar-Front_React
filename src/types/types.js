/** @format */

export const types = {
	uiOpenModal: '[UI] Open Modal',
	uiCloseModal: '[UI] Close Modal',

	eventAddNew: '[Event] Add New',
	eventSetActive: '[Event] Set Active',
	eventClearActiveEvent: '[Event] Clear Active Event',
	eventUpdated: '[Event] Event Updated',
	eventDeleted: '[Event] Event Deleted',
};
