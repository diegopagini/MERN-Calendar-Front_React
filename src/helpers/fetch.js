/** @format */

const baseUrl = process.env.REACT_APP_API_URL;

export const fetchSinToken = (endpoint, data, method = 'GET') => {
	const url = `${baseUrl}/${endpoint}`;
	if (method === 'GET') {
		return fetch(url);
	} else {
		return fetch(url, {
			method: method,
			headers: {
				'Content-type': 'application/json',
			},
			body: JSON.stringify(data),
		});
	}
};
