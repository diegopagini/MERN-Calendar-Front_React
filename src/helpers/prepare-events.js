/** @format */
import moment from 'moment';

export const prepareEvents = (events = []) => {
	return events.map((e) => ({
		...e,
		end: moment(e.end).toDate(),
		start: moment(e.start).toDate(),
	}));
};
