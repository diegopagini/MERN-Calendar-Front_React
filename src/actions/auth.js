/** @format */
import { fetchSinToken, fetchConToken } from '../helpers/fetch';
import { types } from '../types/types';
import Swal from 'sweetalert2';

export const startLogin = (email, password) => {
	return async (dispatch) => {
		const resp = await fetchSinToken('auth', { email, password }, 'POST');
		const body = await resp.json();

		if (body.ok) {
			localStorage.setItem('token', body.token);
			localStorage.setItem('token-init-date', new Date().getTime());

			dispatch(
				login({
					uid: body.uid,
					name: body.name,
				})
			);
		} else {
			Swal.fire('Error', body.msg, 'error');
		}
	};
};

export const startRegister = (email, password, name) => {
	return async (dispatch) => {
		const resp = await fetchSinToken(
			'auth/new',
			{ email, password, name },
			'POST'
		);
		const body = await resp.json();

		if (body.ok) {
			localStorage.setItem('token', body.token);
			localStorage.setItem('token-init-date', new Date().getTime());

			dispatch(
				login({
					uid: body.uid,
					name: body.name,
				})
			);
		} else {
			Swal.fire('Error', body.msg, 'error');
		}
	};
};

export const startChecking = () => {
	return async (dispatch) => {
		const resp = await fetchConToken('auth/renew');
		const body = await resp.json();

		if (body.ok) {
			localStorage.setItem('token', body.token);
			localStorage.setItem('token-init-date', new Date().getTime());

			dispatch(
				login({
					uid: body.uid,
					name: body.name,
				})
			);
		} else {
			dispatch(checkingFinish());
		}
	};
};

const login = (user) => ({
	type: types.authLogin,
	payload: user,
});

const checkingFinish = () => ({
	type: types.authCheckingFinish,
});

export const startLogout = () => {
	return (dispatch) => {
		localStorage.clear();
		dispatch(eventLogout());
		dispatch(logout());
	};
};

const logout = () => ({
	type: types.authLogout,
});

export const eventLogout = () => ({
	type: types.eventLogout,
});
