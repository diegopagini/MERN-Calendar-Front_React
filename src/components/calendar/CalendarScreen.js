/** @format */

import React from 'react';
import { Calendar, momentLocalizer } from 'react-big-calendar';
import 'react-big-calendar/lib/css/react-big-calendar.css';
import moment from 'moment';
import 'moment/locale/es';
import { messages } from '../../helpers/calendar-messages-es';
import { Navbar } from '../ui/Navbar';

moment.locale('es');

const localizer = momentLocalizer(moment);
const events = [
	{
		title: 'CumpleaÃ±os del jefe',
		start: moment().toDate(),
		end: moment().add(2, 'hours').toDate(),
		bgcolor: '#fafafa',
		notes: 'Comprar el pastel',
	},
];

export const CalendarScreen = () => {
	const eventStyleGetter = (evenet, start, end, isSelected) => {
		const style = {
			backgroundColor: '#367cf7',
			bordeRadius: '0px',
			opacity: 0.8,
			display: 'block',
			color: 'white',
		};
		return {
			style,
		};
	};

	return (
		<div className='calendar-screen'>
			<Navbar />
			<Calendar
				localizer={localizer}
				events={events}
				startAccessor='start'
				endAccessor='end'
				messages={messages}
				eventPropGetter={eventStyleGetter}
			/>
		</div>
	);
};
