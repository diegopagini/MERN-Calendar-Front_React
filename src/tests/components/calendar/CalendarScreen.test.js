/** @format */

import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { act } from '@testing-library/react';

import '@testing-library/jest-dom';

import { CalendarScreen } from '../../../components/calendar/CalendarScreen';
import { messages } from '../../../helpers/calendar-messages-es';
import { types } from '../../../types/types';
import { eventSetActive } from '../../../actions/events';

jest.mock('../../../actions/events', () => ({
	eventSetActive: jest.fn(),
	eventStartLoading: jest.fn(),
}));
Storage.prototype.setItem = jest.fn();

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
	calendar: {
		events: [],
	},
	auth: {
		uid: '123',
		name: 'Fernando',
	},
	ui: {
		openModal: false,
	},
};

const store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
	<Provider store={store}>
		<CalendarScreen />
	</Provider>
);

describe('Pruebas en <CalendarScreen />', () => {
	test('debe de mostrarse correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('pruebas con las interacciones del calendario', () => {
		const calendar = wrapper.find('Calendar');

		const calendarMessages = calendar.prop('messages');
		expect(calendarMessages).toEqual(messages);

		calendar.prop('onDoubleClickEvent')();
		expect(store.dispatch).toHaveBeenCalledWith({ type: types.uiOpenModal });

		calendar.prop('onSelectEvent')({ start: 'Hola' });
		expect(eventSetActive).toHaveBeenCalledWith({ start: 'Hola' });

		act(() => {
			calendar.prop('onView')('week');
			expect(localStorage.setItem).toHaveBeenCalledWith('lastView', 'week');
		});
	});
});
