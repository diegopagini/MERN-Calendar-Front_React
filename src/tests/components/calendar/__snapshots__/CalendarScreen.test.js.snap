// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CalendarScreen /> debe de mostrarse correctamente 1`] = `
<div
  className="calendar-screen"
>
  <div
    className="navbar navbar-dark bg-dark mb-4"
  >
    <span
      className="navbar-brand"
    >
      Fernando
    </span>
    <button
      className="btn btn-outline-danger"
      onClick={[Function]}
    >
      <i
        className="fas fa-sign-out-alt"
      />
      <span>
         Salir
      </span>
    </button>
  </div>
  <div
    className="rbc-calendar"
  >
    <div
      className="rbc-toolbar"
    >
      <span
        className="rbc-btn-group"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Hoy
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          &lt;
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          &gt;
        </button>
      </span>
      <span
        className="rbc-toolbar-label"
      >
        junio 2020
      </span>
      <span
        className="rbc-btn-group"
      >
        <button
          className="rbc-active"
          key="month"
          onClick={[Function]}
          type="button"
        >
          Mes
        </button>
        <button
          className=""
          key="week"
          onClick={[Function]}
          type="button"
        >
          Semana
        </button>
        <button
          className=""
          key="day"
          onClick={[Function]}
          type="button"
        >
          Día
        </button>
        <button
          className=""
          key="agenda"
          onClick={[Function]}
          type="button"
        >
          Agenda
        </button>
      </span>
    </div>
    <div
      className="rbc-month-view"
    >
      <div
        className="rbc-row rbc-month-header"
      >
        <div
          className="rbc-header"
          key="header_0"
        >
          <span>
            lun.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_1"
        >
          <span>
            mar.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_2"
        >
          <span>
            mié.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_3"
        >
          <span>
            jue.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_4"
        >
          <span>
            vie.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_5"
        >
          <span>
            sáb.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_6"
        >
          <span>
            dom.
          </span>
        </div>
      </div>
      <div
        className="rbc-month-row"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                01
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                02
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                03
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                04
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                05
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                06
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                07
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                08
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                09
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                10
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                11
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                12
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                13
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                14
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg rbc-today"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                15
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-now rbc-current"
              key="header_1"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                16
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                17
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                18
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                19
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                20
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                21
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                22
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                23
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                24
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                25
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                26
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                27
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                28
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
        </div>
        <div
          className="rbc-row-content"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                29
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                30
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_2"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                01
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_3"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                02
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_4"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                03
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_5"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                04
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_6"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                05
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="btn btn-primary fab"
    onClick={[Function]}
  >
    <i
      className="fas fa-plus"
    />
  </button>
  <Portal
    containerInfo={
      <div
        class="ReactModalPortal"
      />
    }
  />
</div>
`;
